poc/icart_hacspec.py

```diff
@@ -0,0 +1,57 @@
+from hacspec.speclib import *
+
+prime = 2**384 - 2**128 - 2**96 + 2**32 - 1
+
+felem_t = refine(nat, lambda x: x < prime)
+affine_t = tuple2(felem_t, felem_t)
+
+@typechecked
+def to_felem(x: nat_t) -> felem_t:
+    return felem_t(nat(x % prime))
+
+
+@typechecked
+def fadd(x: felem_t, y: felem_t) -> felem_t:
+    return to_felem(x + y)
+
+
+@typechecked
+def fsub(x: felem_t, y: felem_t) -> felem_t:
+    return to_felem(x - y)
+
+
+@typechecked
+def fmul(x: felem_t, y: felem_t) -> felem_t:
+    return to_felem(x * y)
+
+
+@typechecked
+def fsqr(x: felem_t) -> felem_t:
+    return to_felem(x * x)
+
+
+@typechecked
+def fexp(x: felem_t, n: nat_t) -> felem_t:
+    return to_felem(pow(x, n, prime))
+
+
+@typechecked
+def finv(x: felem_t) -> felem_t:
+    return to_felem(pow(x, prime-2, prime))
+
+a384 = to_felem(prime - 3)
+b384 = to_felem(27580193559959705877849011840389048093056905856361568521428707301988689241309860865136260764883745107765439761230575)
+
+def map2p384(u:felem_t) -> affine_t:
+    v = fmul(fsub(fmul(to_felem(3), a384), fexp(u, 4)), finv(fmul(to_felem(6), u)))
+    u2 = fmul(fexp(u, 6), finv(to_felem(27)))
+    x = fsub(fsqr(v), b384)
+    x = fsub(x, u2)
+    x = fexp(x, (2 * prime - 1) // 3)
+    x = fadd(x, fmul(fsqr(u), finv(to_felem(3))))
+    y = fadd(fmul(u, x), v)
+    return (x, y)
+
+inputs = [1, 7, 13, 1<<7, 1<<8, 1<<64, 1<<64-1, prime-1, prime+1]
```

Comment:
Raw Python lists aren't allowed in hacspec. You should wrap it into an array (`array([...])`).

Reply:
Noted -- I'm going to remove this before merging anyway.